current_range_monitor.py

```diff
@@ -1,11 +1,33 @@
+"""
+this is the core app to monitor current range stats
+"""
+
 from itertools import groupby
-from turtle import st
+from current_sensors.sensor import Sensor
+
+
+# note-to-reviewer ->  inorder to not touch the existing tests & app implementaion, I came up with this solution
+def convert_measurements_to_amps(func):
+    """
+    this is a decorator function that converts measurement inputs from sensors to Amps
+    Any function that use this decorator need not worry about the input measurement conversion
+    The sensors should implement the 'Sensor' interface for this to work
 
+    """
+    def convert_to_amps(range):
+        return (
+            func(range.convert_to_amps()) if isinstance(range, Sensor) else func(range)
```

Comment:
Is it possible to avoid `isinstance` here? The issue with its usage is the following: Next, we may have other types and subtypes coming up. Then the `if` will keep getting more complex. See if you can abstract the functionality in a more straight-forward way.